server/rabbitmq.go

```diff
@@ -59,6 +63,7 @@ func (s *AMQPServer) NotifyConnected() chan bool {
 }
 
 func (s *AMQPServer) RegisterEndpoint(endpoint *Endpoint) {
+	endpoint.server = s
```

Comment:
If `server` is not set, the endpoint is invalid and might cause a nil pointer dereference in authenticateEndpointAccess. 

As endpoints get increasingly complex, I'd consider adding an endpoint validation step to the `Register` method and potentially returning an error here or on server.Run